### DISCUSSION TITLE
After adding a route to vue3 router. addRoute(), why is the page menu not updated?

### DISCUSSION THREAD

**Discussion by wenlongFam** [vuejs/rfcs#466] [General]
console.log(router.getRoutes()); The number of arrays is increased by 1, and the menu is not updated

`console.log(router.getRoutes());    //This array is 29，

//add route
  router.addRoute('xitong2', {
    path: '/projectbrowsing/duolianjixitong/xitong2',//
    component: () => import('@/views/xitong/systemTabShow.vue'),
    name: 'Xitong2',
    meta: {
      title: t('routers.xtyc'),
      query: {
        systemId: 'a02'
      }
    }
  })

console.log(router.getRoutes());    //This array is 30，`

**Prompt
Note that adding routes does not trigger new navigation. That is, the added route will not be displayed unless a new navigation is triggered**

Does triggering a new navigation refer to a page jump or something?

**Comment by posva** [comment]
this repo is not for questions. Use the channels mentioned at https://github.com/vuejs/router/issues/1564#issuecomment-1261814099
example of new navigation: `router.push()`
